### DISCUSSION TITLE
GNNExplainer tutorial

### DISCUSSION THREAD

**Discussion by traopia** [pyg-team/pytorch_geometric#6757] [Q&A]
Hello!
I am having some troubles with the [ GNN Explainer example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/gnn_explainer.py)  (examples/gnn_explainer.py): I think that it might be because the example is not up to date but I am not very certain about it. 
Indeed if I call the Explainer Class as specified in the example and in the [documentation tutorial](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/explain.html) : 
`explainer = Explainer(
    model=model,
    algorithm=GNNExplainer(epochs=200),
    explanation_type='model',
    node_mask_type='attributes',
    edge_mask_type='object',
    model_config=dict(
        mode='multiclass_classification',
        task_level='node',
        return_type='log_probs',  # Model returns log probabilities.
    ),
)
`

I get the following error: `__init__() got an unexpected keyword argument 'explanation_type'`
which I fixed by checking the [source code](https://pytorch-geometric.readthedocs.io/en/stable/_modules/torch_geometric/explain/explainer.html#Explainer) and thus calling the Explainer class like this:
`explainer = Explainer(
    model=model,
    algorithm=GNNExplainer(epochs=200),

    model_config=dict(
        mode = 'classification',
        task_level='node',
        return_type='log_probs',
    ),
    explainer_config=dict(explanation_type='model', 
                             node_mask_type='attributes',
                            edge_mask_type='object'),
)`

Now python recognize Explainer to be a torch_geometric.explain.explainer.Explainer type object, but  it doesn't have the functions that I would expect it to have according to the tutorial, such as `explanation.visualize_feature_importance(top_k=10)` or `explanation.visualize_graph()` 
, where explanation was how we instantiate the object Explainer - as indeed I get the error 'GlobalStorage' object has no attribute 'visualize_graph'.  By looking at the source code I found out that those functions are defined for the class [Explanation](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/explain/explanation.html) , but I am not sure if the class Explainer is inheriting those, or at least it seems to me like this is not the case.

Thank you for your help!

**Comment by rusty1s** [comment]
Yeah, we made some interface changes in the new explainer interface. Try the example for PyG 2.2 or update your PyG to latest master. This should resolve it.

**Reply by traopia** [reply]
Thank you for your reply!
Where can I find the example for PyG 2.2? And to which documentation should I thus refer to?

**Reply by rusty1s** [reply]
Take a look at https://github.com/pyg-team/pytorch_geometric/tree/2.2.0/examples and https://pytorch-geometric.readthedocs.io/en/stable/
